Replace drupal-check by Upgrade Status + Drupal Rector

davidferlay opened the pull request:
drupal-check is going deprecated as maintainer stated at https://github.com/mglaman/drupal-check/issues/172#issue-616031763 in favor of tools like [Upgrade Status](https://www.drupal.org/project/upgrade_status)
Behind the scenes Upgrade status uses the same phpstam rules so this module is like an extended version of drupal-check

This PR is also including [Drupal Rector](https://www.drupal.org/project/rector) which helps for d9 readiness.

piggito commented:
Drupal Rector is generating PHP Warnings like these
`PHP Notice:  iconv(): Wrong charset, conversion from `UTF-8' to `ASCII//TRANSLIT//IGNORE' is not allowed in phar:///var/www/html/vendor/rector/rector-prefixed/rector/vendor/nette/utils/src/Utils/Strings.php on line 146
Notice: iconv(): Wrong charset, conversion from `UTF-8' to `ASCII//TRANSLIT//IGNORE' is not allowed in phar:///var/www/html/vendor/rector/rector-prefixed/rector/vendor/nette/utils/src/Utils/Strings.php on line 146
`
I found a workaround at https://github.com/nette/utils/issues/191#issuecomment-502767836 but the root cause of this issue seems to come from compiled code at https://github.com/rectorphp/rector-prefixed rather than nettle package.

andypost commented:
So it could use to install (require-dev) dgo.to/upgrade_rector and dgo.yo/, upgrade_status

davidferlay commented on scripts/makefile/tests.mk:
```diff
@@ -86,12 +86,19 @@ else
-	$(call php, vendor/bin/drupal-check -V)
-	$(call php, vendor/bin/drupal-check -ad -vv -n --no-progress web/modules/custom/)
+	$(call php, vendor/bin/rector -V)
+	$(call php, vendor/bin/rector process --dry-run --no-progress-bar web/modules/custom web/themes/custom)
```
If rector.yml file is required for script to run, best to wrap this around code to check file existence (check example in scripts/makefile/tests.mk)

piggito replied:
Added validation for file to exist

davidferlay replied:
Can't see where : o

davidferlay replied:
ah found but looks like misplaced : d

![image](https://user-images.githubusercontent.com/20738264/83269953-18cf6e80-a1c8-11ea-86cf-46d1c831d96e.png)

piggito replied:
Fixed

davidferlay commented:
Can you please also add new makefile targets in here ? https://github.com/skilld-labs/skilld-docker-container/blob/a6615f35af4280df3ff2ff175e5eff333a721023/scripts/makefile/tests.mk#L168

davidferlay commented:
Added disabling of modules after test and updated error message of upgrade-status a bit : 

![image](https://user-images.githubusercontent.com/20738264/83269268-26d0bf80-a1c7-11ea-8123-249eec3560de.png)
